File: project_euler/problem15.rb
def factorial(num)
  if num == 0
  	1
  else
    num * factorial(num-1)
  end
end

def combination(size_of_grid)
  factorial(size_of_grid * 2) / (factorial(size_of_grid) * factorial(size_of_grid))
    
end

p combination(20)

File: problem15.py
def factorial(num):
  if (num == 0):
    return 1
  return reduce(lambda x,y: x * y, range(1,num + 1))

def number_of_paths(grid_size):
  return factorial(grid_size * 2)/ (factorial(grid_size) ** 2)

print number_of_paths(20)
